### ISSUE TITLE
[Proposal] Remove `tests/**/output` from vcs

### ISSUE THREAD

**Issue by sneakers-the-rat** [linkml/linkml#1958]
**Is your feature request related to a problem? Please describe.**

Whenever I run tests locally, I always end up with a lot of the files in various `output` folders changed, and then I need to be careful to ignore them when committing, or else run `git update-index --assume-unchanged` for all of them a la here: https://github.com/linkml/linkml/blob/0c3afa90701dd18ce874f6103cc7668684eb697d/hide_test_changes.sh#L3

that call doesn't persist across switching branches, which one needs to do all the time, so it's like playing whack-a-mole where inevitably i will end up committing one of them.

I am not quite sure what the purpose of having them versioned is - being able to look at the intended outputs all the time is nice, and in that case we'd expect those files to not be used in the tests since the test should be self-contained. Using those as expected outputs to compare tests to would also be good, but then since the tests also generate there then their status becomes ambiguous. 

It seems like `__snapshot__` is supposed to be the way to store expected outputs that are used in the tests, but many tests use `outputs` instead - eg. any test that uses `TestEnvironment` does, since `outdir` is `output` : https://github.com/linkml/linkml/blob/0c3afa90701dd18ce874f6103cc7668684eb697d/tests/utils/test_environment.py#L42

which is used by `expected_path`: 
https://github.com/linkml/linkml/blob/0c3afa90701dd18ce874f6103cc7668684eb697d/tests/utils/test_environment.py#L97

which is used in turn by `TestEnvironment.generate_single_file`, `GeneratorTestCase.single_file_generator`, which is used in `test_base` and `test_biolink_model` . 

So ultimately it isn't clear to me what the `output` directories are for, and seems like it has the potential to cause weird testing errors and make PRs more difficult to review. 

**Describe the solution you'd like**
I think we should remove `output` directories from being tracked by git and make a clearer distinction between the types of test directories. I like the way that test directory structure is mirrored across the test packages, so this hopefully shouldn't be too hard.
- `__snapshots__`  (versioned) - frozen output used to test
- `temp` (ignored) - output that is generated during testing that is cleared during cleanup, ideally by a fixture: if one wants to keep those outputs, we already have a `--with-output` flag for that.
- `examples` (versioned) - sample output that is used to be able to see expected results without them being used in testing, if desired.

We can then clarify in the developer docs what the directory structure is within the tests, as well as document how the various utility classes and fixtures are intended to be used.

Currently, `output/` is in `.gitignore` already, so new files generated by tests aren't versioned, which also leads to inconsistencies eg. if we have new tests that use shared fixtured, but also means that it shouldn't be too big of a change to remove them from tracking

**How important is this feature?** Select from the options below:
• Low - it's an enhancement but not crucial for work

**When will use cases depending on this become relevant?** Select from the options below:
(continuously, doesn't have a timescale)

**Comment by cmungall** [comment]
Yes, this a longstanding ergonomic issue. We have started moving from the older check-in `output` folder strategy to the snapshot strategy, but this is incomplete. I'm in favor of simply doing one-time removal of `output`s from git (and they won't come back thanks to `.gitignore`)
